src/ucs/async/signal.h

```diff
@@ -35,4 +38,18 @@ typedef struct ucs_async_signal_context {
         --(_async)->signal.block_count; \
     }
 
+#ifndef SIGEV_THREAD_ID
+#define SIGEV_THREAD_ID 4
+#endif
+
+struct ksigevent {
```

Comment:
maybe define it only if there is no sigevent on the system?

Reply:
Nope. The definition of sigevent varies across Linux libc implementations; this is the fundamental problem. In particular, usage with `SIGEV_THREAD_ID` isn't standardized, so the only pragmatic approach is to try to match the kernel's definitions. Of course, one could try to pull in the kernel headers, but that often causes subtle (and not so subtle) clashes of definition.